src/ape_cache/query.py

```diff
@@ -239,15 +239,19 @@ def perform_contract_event_clause(self, query: ContractEventQuery) -> TextClause
         )
 
     def perform_query(self, query: QueryType) -> pd.DataFrame:
-        with self.database_connection as conn:
-            result = conn.execute(self.perform_query_clause(query))
+        try:
```

Comment:
So, the thinking here is that `estimate_query` should return `None` in any case that there would be an exception here, otherwise the query interface will not function properly. I'd like to keep that this raises an exception here, because it shows that the estimation logic is not performing correctly and return `None` where it should be.

Reply:
`estimate_query` logs and then returns `None` in the exception. So I think this functions as we would like